odoo/tools/sql.py

```diff
@@ -313,25 +315,34 @@ def reverse_order(order):
     return ', '.join(items)
 
 
-def increment_field_skiplock(record, field):
+def increment_fields_skiplock(records, *record_fields):
     """
-        Increment 'friendly' the [field] of the current [record](s)
+        Increment 'friendly' the given field(s) of the current record(s)
         If record is locked, we just skip the update.
         It doesn't invalidate the cache since the update is not critical.
 
-        :rtype: bool - if field has been incremented or not
+        :param records: recordset to update
+        :param record_fields: integer fields to increment
+        :returns: whether the specified fields were incremented on any record
+        :rtype: bool
     """
-    if not record:
+    if not records:
         return False
 
-    assert record._fields[field].type == 'integer'
+    for field in record_fields:
+        assert records._fields[field].type == 'integer'
 
-    cr = record._cr
-    query = """
-        UPDATE {table} SET {field} = {field} + 1 WHERE id IN (
-            SELECT id from {table} WHERE id in %(ids)s FOR UPDATE SKIP LOCKED
-        ) RETURNING id
-    """.format(table=record._table, field=field)
-    cr.execute(query, {'ids': tuple(record.ids)})
+    field_identifiers = tuple(map(Identifier, record_fields))
 
-    return bool(cr.fetchone())
+    query = SQL("""
+        UPDATE {table}
+           SET ({fields}) = ROW({fields_plus_one})
+         WHERE id IN (SELECT id FROM {table} WHERE id in %(ids)s FOR UPDATE SKIP LOCKED)
+    """).format(
+        table=Identifier(records._table),
+        fields=SQL(',').join(field_identifiers),
+        fields_plus_one=SQL(',').join(field + SQL(' + 1') for field in field_identifiers)
+    )
+    cr = records._cr
+    cr.execute(query, {'ids': tuple(records.ids)})
```

Comment:
Alternatively, replace `in` by `= any(%(ids)s)`, and pass in a list directly (so no need for the list -> tuple conversion here).

Though it doesn't apply here, generally for values I tend to favor `= any()` as it allows empty collections, whereas `in` does not.